### ISSUE TITLE
[Bug] UINavigationBar.Appearance.BarTintColor not working when update 4.5 version

### ISSUE THREAD

**Issue by haiduong741** [xamarin/Xamarin.Forms#9765]
### Steps to Reproduce

1. In Xamarin.iOS project. Added this code 
` UINavigationBar.Appearance.BarTintColor = UIColor.FromPatternImage(UIImage.FromFile("toolbarBackground.png"));
            UINavigationBar.Appearance.TintColor = UIColor.White;
            UINavigationBar.Appearance.SetTitleTextAttributes(new UITextAttributes() { TextColor = UIColor.White });` 
2. Build and Run

### Expected Behavior
![image](https://user-images.githubusercontent.com/37478492/75502632-fcc00380-5a05-11ea-8a09-35652b4abe83.png)
### Actual Behavior
![image](https://user-images.githubusercontent.com/37478492/75502775-6809d580-5a06-11ea-9650-dcd908a4444d.png)

### Basic Information

- Version with issue: 4.5
- Last known good version: 4.4
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Screenshots
![image](https://user-images.githubusercontent.com/37478492/75502802-7ce66900-5a06-11ea-8344-706cd6819d9b.png)

**Comment by rclancy** [comment]
I'm experiencing exactly the same issue, and it only started today. Yesterday the colors were there.

**Comment by rclancy** [comment]
I downgraded Xamarin.Forms to 4.4 and the problem went away. This morning I upgraded to 4.5. That seems to have caused the issue.

**Comment by SandraHicks** [comment]
Same issue here just after upgrading to 4.5.0.356.
Trying to set this code in AppDelegate FinishedLaunching:
UINavigationBar.Appearance.BarTintColor = UIColor.FromPatternImage(UIImage.FromFile("toolbar_image.png"));

No exception, just nothing showing up in the toolbar as in the screenshots of haiduong741.

**Comment by enterprisemk** [comment]
Same issue here. UINavigationBar.Appearance.BarTintColor not working.

Downgrade to v4.4 and issue goes away.

**Comment by GalaxiaGuy** [comment]
My guess would be this PR: https://github.com/xamarin/Xamarin.Forms/pull/9346

Out of curiosity, does the behavior change between iOS 13 and iOS <13?

**Comment by jsuarezruiz** [comment]
Seems to be the same (or related) as https://github.com/xamarin/Xamarin.Forms/issues/9767

**Comment by hartez** [comment]
Duplicate of #9790

**Comment by PawKanarek** [comment]
Any workaround for this? i need to go on production and i have already implemented some features from 4.5, and i don't want to revert all changes, because my last Xamarin.Forms was 4.3. Any Hacky CustomRenderer wouldn't fix this?

**Comment by haiduong741** [comment]
XF 4.5.0.396 doesn't fixed this issue. :(

**Comment by PawKanarek** [comment]
So how do i release new app with new X.F on mac after april? (The depracated UIWebView issue)  With this ugly app bar?

**Comment by ngalicoco** [comment]
Please when is this coming out. It is not fixed in 4.5.0.396 ? Any other alternative, like setting something in custom renderer will fix this ? Can't till 4.7

**Comment by kcrg** [comment]
Please fix asap, it's urgent.

**Comment by PawKanarek** [comment]
Guys? The April is near...

**Comment by haiduong741** [comment]
> Guys? The April is near...

Yes. April is comming. And I must choose between 2 options XF 4.5 remove UIWebView and 4.4 using Navigation has background.

**Comment by PawKanarek** [comment]
But i don't have an option as one of my App is not published. So i have only option with XF 4.5 and this navigation bar. My QA and PM are asking me everyday about this bug because I have 4 buttons on navigation bar. (LeftItem for LoggingIn/Settings, Right buttons for Seach, Chromecast, and Secondary Items). Also colors of this navigation bar can be changed, so there are 5 things that this bug is causing :(

**Comment by samhouts** [comment]
I'm closing this because it is a duplicate of #9790. That issue has been resolved, and the fix will be released in a 4.5.0 service release today or tomorrow. Thanks for your patience!

**Comment by ngalicoco** [comment]
> I'm closing this because it is a duplicate of #9790. That issue has been resolved, and the fix will be released in a 4.5.0 service release today or tomorrow. Thanks for your patience!

@samhouts If possible, can you please post here once is the fix is available on nuget as we are all following this conversation ?

**Comment by samhouts** [comment]
It should be available now! https://www.nuget.org/packages/Xamarin.Forms/4.5.0.495

**Comment by eevahr** [comment]
4.5.0.495 does not seem to resolve the issue for me...

**Comment by enterprisemk** [comment]
4.5.0.495 - still not working for me

**Comment by ngalicoco** [comment]
Same here 4.5.0.495 - still not working for me

**Comment by haiduong741** [comment]
> It should be available now! https://www.nuget.org/packages/Xamarin.Forms/4.5.0.495

why you close this when XF not fixed this issue in new version???? 
*ridiculous

> I'm closing this because it is a duplicate of #9790. That issue has been resolved, and the fix will be released in a 4.5.0 service release today or tomorrow. Thanks for your patience!
"and the fix will be released in a 4.5.0"
*ridiculous 💃

**Comment by zuimengaitianya** [comment]
4.5.0.495 - still not working for me

**Comment by eevahr** [comment]
Hi @samhouts do you have any updates on this issue? :)

**Comment by dpozimski** [comment]
Hi, I have the same issue too.
We have to release 4.5 version (with new web view) but we cant because our navigation bar is broken...

**Comment by samhouts** [comment]
@eevahr Thanks for the ping! We don't always see conversation on closed issues unless there is a specific ping, so I missed the comments saying this wasn't working. 

I'll reopen this, and we'll take a look. Thanks!

**Comment by eevahr** [comment]
Hola, is it possible to get a more spesific timeframe on this, i am really aching for an ios update and would love to know if this fix is going to be released in the next 24 hours or less. As you know as of today 4.4 is not an option 😬

**Comment by jfversluis** [comment]
If you have a version already in the store then you have until December regarding the `UIWebView` deprecation, so that should be no issue. We have a PR open for this issue now which will hopefully be merged soon. A release should then follow soon.

**Comment by eevahr** [comment]
Nice, that's a load off, i totally missed that from the apple notice, Thanks!

**Comment by PawKanarek** [comment]
You are lucky guy then :) One of my company's app is blocked by this right now :D 👍 🥇

**Comment by samhouts** [comment]
Please test with this package and let us know if your issue is still not resolved. Thanks!

[Xamarin.Forms.4.5.0.1682.nupkg.zip](https://github.com/xamarin/Xamarin.Forms/files/4422303/Xamarin.Forms.4.5.0.1682.nupkg.zip)

**Comment by haiduong741** [comment]
> Please test with this package and let us know if your issue is still not resolved. Thanks!
> 
> [Xamarin.Forms.4.5.0.1682.nupkg.zip](https://github.com/xamarin/Xamarin.Forms/files/4422303/Xamarin.Forms.4.5.0.1682.nupkg.zip)

Thanks for your works. Im always following this issue. Tomorrow I will check this.

**Comment by haiduong741** [comment]
@samhouts 
I was add your package. But it not fixed. :( 
![image](https://user-images.githubusercontent.com/37478492/78315305-ce928e00-7586-11ea-8791-e7d28e2e511f.png)
![image](https://user-images.githubusercontent.com/37478492/78315387-08639480-7587-11ea-86dc-270728ea277f.png)

**Comment by haiduong741** [comment]
FYI
Set background Image it work fine. But when I set background gradient using custom NavigationPageRenderer it not show. Here my code

```
using CoreAnimation;
using CoreGraphics;
using Demo;
using UIKit;
using Xamarin.Forms;
using Xamarin.Forms.Platform.iOS;

[assembly: ExportRenderer(typeof(NavigationPage), typeof(CustomNavigationPageRenderer))]
namespace Demo
{
    public class CustomNavigationPageRenderer : NavigationRenderer
    {
        public FNavigationPageRenderer() : base()
        {
        }

        public override void ViewWillAppear(bool animated)
        {
            base.ViewWillAppear(animated);
            UpdateGradientView();
        }

        private void UpdateGradientView()
        {
            var control = this.Element as FNavigationPage;
            var gradientLayer = new CAGradientLayer();
            gradientLayer.Bounds = NavigationBar.Bounds;
            gradientLayer.Colors = new CGColor[] { control.RightColor.ToCGColor(), control.LeftColor.ToCGColor() };
            gradientLayer.EndPoint = new CGPoint(0.0, 0.5);
            gradientLayer.StartPoint = new CGPoint(1.0, 0.5);
            UIGraphics.BeginImageContext(gradientLayer.Bounds.Size);
            gradientLayer.RenderInContext(UIGraphics.GetCurrentContext());
            UIImage image = UIGraphics.GetImageFromCurrentImageContext();
            UIGraphics.EndImageContext();
            NavigationBar.SetBackgroundImage(image, UIBarMetrics.Default);
        }
    }
}
```

**Comment by haiduong741** [comment]
FYI
Set background Image it work fine. But when I set background gradient using custom NavigationPageRenderer it not show. Here my code
```

using CoreAnimation;
using CoreGraphics;
using FastMobile.Device.iOS;
using UIKit;
using Xamarin.Forms;
using Xamarin.Forms.Platform.iOS;

[assembly: ExportRenderer(typeof(NavigationPage), typeof(CustomNavigationPageRenderer))]
namespace FastMobile.Device.iOS
{
    public class CustomNavigationPageRenderer : NavigationPageRenderer
    {
        public CustomNavigationPageRenderer() : base()
        {
        }

        public override void ViewWillAppear(bool animated)
        {
            base.ViewWillAppear(animated);
            UpdateGradientView();
        }

        private void UpdateGradientView()
        {
            var control = this.Element as FNavigationPage;
            var gradientLayer = new CAGradientLayer();
            gradientLayer.Bounds = NavigationBar.Bounds;
            gradientLayer.Colors = new CGColor[] { control.RightColor.ToCGColor(), control.LeftColor.ToCGColor() };
            gradientLayer.EndPoint = new CGPoint(0.0, 0.5);
            gradientLayer.StartPoint = new CGPoint(1.0, 0.5);
            UIGraphics.BeginImageContext(gradientLayer.Bounds.Size);
            gradientLayer.RenderInContext(UIGraphics.GetCurrentContext());
            UIImage image = UIGraphics.GetImageFromCurrentImageContext();
            UIGraphics.EndImageContext();
            NavigationBar.SetBackgroundImage(image, UIBarMetrics.Default);
        }
    }
}
```

**Comment by haiduong741** [comment]
> If you have a version already in the store then you have until December regarding the `UIWebView` deprecation, so that should be no issue. We have a PR open for this issue now which will hopefully be merged soon. A release should then follow soon.

My app is new app. So sad :(

**Comment by ngalicoco** [comment]
@haiduong741 I found a workaround, not sure if this helps you. When instantiaing a new Navigation Page you can set the BarBackGroundColor and BarTextColor. Although if you have a Tint Color, Xamarin deprecated that property.
`var page= new NavigationPage(page)
            {
                Title = "Test Navigation Page",
                IconImageSource = "Test.png",
                BarBackgroundColor = Color.Blue,
                BarTextColor = Color.White
            };`
But if you have more than one Navigation Page and you want to set this on all pages, you can write a custom NavigationRenderer and apply the barbackground color and bar text color for the element which is a NavigationPage.
Hope this helps and also Xamarin fixes the issue soon, as it is not just UIWebview issue, there are other important issues they fixed with 4.5, and we need them too.

**Comment by haiduong741** [comment]
> @haiduong741 I found a workaround, not sure if this helps you. When instantiaing a new Navigation Page you can set the BarBackGroundColor and BarTextColor. Although if you have a Tint Color, Xamarin deprecated that property.
> `var page= new NavigationPage(page) { Title = "Test Navigation Page", IconImageSource = "Test.png", BarBackgroundColor = Color.Blue, BarTextColor = Color.White };`
> But if you have more than one Navigation Page and you want to set this on all pages, you can write a custom NavigationRenderer and apply the barbackground color and bar text color for the element which is a NavigationPage.
> Hope this helps and also Xamarin fixes the issue soon, as it is not just UIWebview issue, there are other important issues they fixed with 4.5, and we need them too.

Thanks for your answer. But this has issue when using custom background Gradient. Solid color is work fine.

**Comment by dmarjoram** [comment]
Not fixed for me in 4.5.0.617

**Comment by eevahr** [comment]
Have you cleaned ```bin/ obj/``` in all projects? I´m on ```4.5.0.530``` and it works fine for me.

**Comment by dmarjoram** [comment]
On iOS 12 and large titles enabled, background image is still not rendered when overriding NavigationRenderer. You just see the white background.

`    public class BackgroundImageNavigationRenderer : NavigationRenderer
    {
        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            View.BackgroundColor = UIColor.Clear;
            NavigationBar.BackgroundColor = UIColor.Clear;
			
            var image = UIImage.FromFile(Xamarin.Forms.Application.Current.Resources["BackgroundImage"] as string);

            NavigationBar.BarTintColor = UIColor.FromPatternImage(image);
		}
	}`
